### ISSUE TITLE
Oversized embedded slides with embed_xaringan()

### ISSUE THREAD

**Issue by spcanelon** [gadenbuie/xaringanExtra#69]
Hi @gadenbuie !

I updated a [blog post today](https://silvia.rbind.io/2020-10-07-rstudio-instructor-certification-tidyverse/#preparing-for-the-exams) with some simple text edits and now my embedded xaringan slides look comically oversized 😆 

Any idea why this might be happening? I don't know if it's xaringanExtra-related or if I should look elsewhere for clues. I'm using xaringanExtra v0.2.4.

Here's a screenshot with the page at 50% magnification 🤣 

![image](https://user-images.githubusercontent.com/49913337/97485986-38162780-1931-11eb-99b3-537c2c05d790.png)

**Comment by gadenbuie** [comment]
Yikes! But I think the fix will be easy. It looks like you might have your blogdown repo in a shared drive (Dropbox or OneDrive, just a guess)? There somehow have been conflicts in the names of one of the dependencies for the _share again_ extension: there are a few version of [fitvids.js in this commit](https://github.com/spcanelon/silvia/commit/613ef8808bf01adb15c861d42b80111266768bdf).

If you delete those extra versions and re-render, you should get the right `fitvids.js` back again and all will be well in the tidyverse.

**Comment by spcanelon** [comment]
Yes! I keep my blogdown repo in a Box drive and I see what you mean about the extra versions of fitvids.min.js. 

I deleted them all and re-rendered so that [now's there's only the one fitvids.min.js](https://github.com/spcanelon/silvia/tree/master/content/post/2020-10-07-rstudio-instructor-certification-tidyverse/index_files/fitvids), but the slides are still giant 🤔

**Comment by spcanelon** [comment]
I found a temporary fix by changing the embed ratio from 16:9 to 4:3. I'll keep digging around to see what I can find out 🔍 

![image](https://user-images.githubusercontent.com/49913337/97514319-c8219480-1964-11eb-89f7-e6ba525abec8.png)

**Comment by gadenbuie** [comment]
So, from looking at the developer console while loading your page, I can see that the browser is failing to load http://silvia.rbind.io/2020-10-07-rstudio-instructor-certification-tidyverse/index_files/fitvids/fitvids.min.js which means that the file isn't being served by Netlify.

I think the issue is that you may have knit the document directly in RStudio. When you render with `blogdown::serve_site()`, the `index_files/` folder should be moved. You'll know that it worked correctly when `index.html` for the post references 

```html
<script src="/rmarkdown-libs/fitvids/fitvids.min.js"></script>
```

at the top of the HTML file. (Currently it references `index_files/...` instead of `/rmarkdown-libs/...`.)

The changes should look [like these changes](https://github.com/spcanelon/silvia/compare/master...gadenbuie:fix-fitvids?expand=1). I can submit that as a PR, but there were quite a lot of other changes that happened when I re-rendered this post and maybe you'll want to take a look at and possibly commit those changes.

**Comment by spcanelon** [comment]
Garrick, thanks so much for helping me sleuth! Specially because the issue has not to do with xaringanExtra 🙈 
This is all really helpful context -- I'll make sure to take a close look at those proposed changes for my next round of commits 👍

**Comment by spcanelon** [comment]
**Update and fix!** 

I hadn't been knitting directly, but in the process of making sure I was using `blogdown::serve_site()` I noticed this message in my console that I hadn't been paying attention to 🤦‍♀️ 

```
--------------------------------------------------------------------------------
You are recommended to ignore more items in the ‘ignoreFiles’ field in config.toml: “\\.knit\\.md$“, “\\.utf8\\.md$”
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
You are recommended to remove the item ‘_files$’ in the ‘ignoreFiles’ field in config.toml.
--------------------------------------------------------------------------------
```

[I followed the 2nd recommendation](https://github.com/spcanelon/silvia/commit/81a4ecfb5b7371e23987cc0b3ce6ad32a8bed85d) and removed `"_files$"` from the `ignoreFiles` field of `config.toml` and problem solved! 

So I guess this means that before I made this change, blogdown was ignoring the entire `index_files` folder being generated with the xaringanExtra libs? I'm still not sure how this suddenly became an issue though, I don't just go around tweaking that field in my `config.toml` 😅 -- nevertheless I'm glad my embedded slides are back to a reasonable size!

**Comment by gadenbuie** [comment]
Yeah that's interesting and I noticed it too. At one point, I thought that was the problem, but then I looked around and the advice is mixed on that. For instance, hugodown [recommends ignoring `_files$`](https://hugodown.r-lib.org/articles/config.html).

On the other hand, there are a quite a few different workflows for rendering blogdown sites, so definitely use what works for you!

**Comment by gadenbuie** [comment]
Oh and recently blogdown introduced a few new features that probably make this exact setup easier: 

> Personal highlights from the latest [\#blogdown](https://twitter.com/hashtag/blogdown)\
> \
> 🧶 the knit button in [\@rstudio](https://twitter.com/rstudio) \*works\*\
> \
> 🍱 has a helper function blogdown::bundle\_site() to move post files into leaf bundles\
> \
> 🏗️ blogdown::build\_site() no longer recompiles R Markdown files by default\
> \
> 🎓 Hugo modules just work <https://t.co/sfqcyrCdnx>
>
> --- Alison Presmanes Hill (\@apreshill) [October 14, 2020](https://twitter.com/apreshill/status/1316441899893493763)

**Comment by spcanelon** [comment]
Ah, good to know -- thanks again!

**Comment by cderv** [comment]
Hi, 

yes blogdown is now using [Page bundle](https://gohugo.io/content-management/page-bundles/) by default and the resources files created by **rmarkdown** in the `_files` folder are now leaving the the same place as the post (in the bundle), and you should not ignore them.

It  was a [change in blogdown 0.21](https://github.com/rstudio/blogdown/blob/9cfef67ee380f855475d9c4a11615a4332125a97/NEWS.md#changes-in-blogdown-version-021)

> For page bundles, the index_files/ and index_cache/ folders are no longer moved to the static/ directory (for other types of posts, these folders are still moved). Consequently, you should not ignore "_files$" in the ignoreFiles field in your config.toml or config.yaml any more.

There are more changes coming into next version that will improve what was started in 0.21. There will be some check functions to help ! We are also working on improving all those messages to make it more effective.
See @apreshill last talk for a preview: https://alison.rbind.io/talk/2020-larug/

Sorry for the disturbance with all these changes, ... but hopefully it is for the best! 😄

**Comment by spcanelon** [comment]
Hi @cderv, thank you for elaborating on this and for including such helpful references in your comment.

I was able to catch Alison's talk live on Tuesday! I'm excited about both the new check functions as well as the formatting/style of the accompanying messages. I expressed my support for these types of user-friendly messages in [blogdown issue #510](https://github.com/rstudio/blogdown/issues/510#issuecomment-743468440) as it's particularly relevant to this current issue
